### ISSUE TITLE
false positives for SCSS function arguments with declaration-property-value-keyword-no-deprecated

### ISSUE THREAD

**Issue by woylie** [stylelint/stylelint#8575]
### What minimal example or steps are needed to reproduce the bug?

```scss
$colors: (
  background: (
    primary: #f00
  )
);

@function mycolor($id, $ids...) {
  @return map.get($colors, $id, $ids...);
}

.some-class {
  background-color: mycolor(background, primary);
}
```

### What minimal configuration is needed to reproduce the bug?

```json
{
  "extends": [
    "stylelint-config-standard-scss"
  ]
}
```

### How did you run Stylelint?

```shell
stylelint css
```

### Which Stylelint-related dependencies are you using?

```json
{
    "stylelint": "16.19.1",
    "stylelint-config-recess-order": "6.0.0",
    "stylelint-config-standard-scss": "15.0.0",
    "stylelint-declaration-strict-value": "1.10.11",
    "stylelint-webpack-plugin": "5.0.1"
}
```

### What did you expect to happen?

No problems being reported for SCSS function arguments. No auto-fixes applied.

### What actually happened?

This warning is emitted:

```
Expected "background" to be "canvas"  declaration-property-value-keyword-no-deprecated
```

The auto-fixer updates the code to:

```scss
.some-class {
  background-color: mycolor(canvas, primary);
}
```

**Comment by jeddy3** [comment]
@woylie Thanks for the report and for using the template.

I can reproduce this using [the demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcICGACYAdAdilkA2EATvCgLSkBmEEAFAEZJgDWA5oRAK6ZQCUA3FgC+IADQgKASzwwAckgC2cRDAAeCgA7SAdGADOu0eAiZJLBCAzYU6EKoAuMbrpskA2lhw4buuwE9peBKYdqSQJhIspD5I3EiEUDYeKAC6QoZhpgBiRPJIduYAVrrGhrDqBoiWXiA+-jCBwS7WIHh5MD42IknefgFBIRkRUXYxUHEJCM2tDh0gaYJAA) and:

```css
a {
  color: --foo(background);
}
```

I've labelled the issue as ready to implement. Please consider [contributing](https://stylelint.io/contributing) if you have time.

There are [steps on how to fix a bug in a rule](https://stylelint.io/developer-guide/rules#fix-a-bug-in-a-rule) in the Developer guide.

The fix will likely involve ignoring keywords inside functions.

**Comment by romainmenke** [comment]
Another case to consider is this:

```css
div {
  color: attr( background type(<color>), lime);
}
```

[Demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcIoEsBuACYAdAdu9kANhAE7zoCGALlSQBToBGFYA1gOYkQCuOU6VAJ4AHGHQA8RUgD4AlABp0hZAFsYsgNy4AviHkgAZskIwAchTUIQMAB4XhJgHRgAzi73gIOI+yvY86FjWNlQwfC5B5ADauPj4QS5CJso4VAC0kN7I7GmJFHwUJFBBsegAujoemT4AYqQq1FaOiYLJyKkkYI4AVi5eHrDC7oj+8SAtbamRgSCE1DCJQfKlCUkwKenV2blU+VCFxQgzc6GLIJXaQA)

Check in Chrome: https://codepen.io/romainmenke/pen/ByzNBaR

The underlying problem in both cases is that we don't actually check if the found keyword actually matches a specific syntax (position).

This should error:

```css
div {
  color: attr( foo type(<color>), background);
}
```

But this must be fine:

```css
div {
  color: attr( background type(<color>), lime);
}
```

The syntax for `attr` is: `attr( <attr-name> <syntax>? , <declaration-value>?)` where `<attr-name>` can be any `ident`.

So any matches in `<attr-name>` should be ignored while anything in `<declaration-value>` should error.

The same can also be done with style queries in the `if()` function. As it is also possible to assign arbitrary keywords to custom props and use style queries to match. e.g. `style(--foo: background)`

**Comment by ragini-pandey** [comment]
Raised a new PR for this issue as there is no activity on draft PR
